kythe/rust/indexer/src/providers.rs

```diff
@@ -91,26 +115,28 @@ impl KzipFileProvider {
 impl FileProvider for KzipFileProvider {
     /// Given a file hash, returns whether the file exists in the kzip.
     fn exists(&mut self, file_hash: &str) -> bool {
-        let name = format!("files/{}", file_hash);
+        // root_name contains a trailing forward-slash, so it's needed in the format
+        // string. For example, the extractor kzip that will have a root name of "root/"
+        let name = format!("{}files/{}", self.root_name, file_hash);
         self.zip_archive.by_name(&name).is_ok()
     }
 
-    /// Given a file hash, returns the string contents of the file from the
+    /// Given a file hash, returns the vector of bytes of the file from the
     /// kzip.
     ///
     /// # Errors
     ///
     /// If the file does not exist in the kzip, a
     /// [FileNotFoundError][KytheError::FileNotFoundError] is returned.
-    /// If the file cannot be read into a string, a
-    /// [FileReadError][KytheError::FileReadError] is returned.
-    fn contents(&mut self, file_hash: &str) -> Result<String, KytheError> {
+    /// If the file cannot be read, a FileReadError][KytheError::FileReadError]
```

Comment:
nit: here you can simply say `if the file can not be read, an error is returned`.  We know what error it is from the type signature.  This saves you some typing, as the `KytheError::FileReadError` will be hyperlinked automatically by any IDE worth its salt, or by rustdoc.

Reply:
So keep the hyperlink, but change the phrasing and just make the word `error` be hyperlinked to the error type instead?

Reply:
You don't need to hyperlink anything here.  Case in point this would be enough:

```
If an error is returned, that means that the file could not be read.
```